R/loadFromMetaphlan.R

```diff
@@ -149,20 +149,27 @@ loadFromMetaphlan <- function(file, sample_meta = NULL, phy_tree = NULL, ...){
 }
 
 # Check that metaphlan file contains correct information
-.check_metaphlan <- function(data){
-    # Get rowdata columns
-    rowdata_columns <- data[ , 1:2]
-    # Get columns that go to assay
-    assay_columns <- data[ , 3:ncol(data)]
+.check_metaphlan <- function(x){
+    # Getting column indices of character columns
+    col_idx <- sapply(x, is.character)
+    # Get rowdata column indices
+    col_rowdata_idx <- col_idx[col_idx==TRUE]
+    # Getting column indices of numeric columns
+    col_idx <- sapply(x[,!col_idx], is.numeric)
```

Comment:
These are not "numeric columns" by definition because col_idx lists character columns, and its complement !col_idx lists non-character columns. The "non-character columns" do not equal to "numeric columns". 

Why not identify character columns with `col_idx <- sapply(x, is.character)` and numeric columns with `col_idx <- sapply(x, is.numeric)` and also check that there are no other types columns.

Reply:
yes they are not numeric columns, they are everything else but character columns.
The idea of `x[,!col_idx]` in sapply was to reduce the amount of checks to the rest of the columns that were not characters (using and saving computing results from first check). But true theoretically metaphlan outputs should have beyond 2-3 character columns, so not much is saved. I will put it to just `x`

Reply:
But if you can safely assume that there are only numeric and character columns then it would be safer to explicitly pick the numeric ones (and throw error if there are other categories)? Then you also do not need to check any more later if the columns are numeric.

Reply:
The idea of the following checks (which I will update soon according to the other comments)
```
# Check presence of other data than numeric and character
    # Check rowdata exist
    # Check assay abundance data exist
    # Check rowdata column names that they contain right information.
    # If these requirements are met, give FALSE. Otherwise, give TRUE.
    if( sum(col_idx==FALSE)==0 && 
        sum(col_rowdata_idx)!=0 && 
        sum(col_assay_idx)!=0 &&
        any(names(col_rowdata_idx) %in% "clade_name") ){
        result <- FALSE
    }
```
1st throw error if some other type than numeric or character is present
2nd presence of at least one column at rowdata
3rd presence of at least one column at numeric data
and presence of "clade_name" among rowdata upon which the rest of the metaphlan parsing will depend on.
or sorry I didn't get what you mean?

Reply:
Ok good. Perhaps @TuomasBorman could also check if he has comments on the implementation of such checks in general.

Reply:
I think "the most important thing" is that there are sufficient documentation; what is the purpose of each check